src/plugins/data/public/query/query_string/language_service/query_result.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import './_recent_query.scss';
+
+import { BehaviorSubject } from 'rxjs';
+import { EuiButtonEmpty, EuiPopover, EuiText, EuiPopoverTitle } from '@elastic/eui';
+
+import React, { useState, useEffect } from 'react';
+
+export interface QueryStatus {
+  status: string;
+  message?: {
+    error?: string;
+    message?: string;
+    statusCode?: number;
+  };
+  time?: number;
+}
+
+export function QueryResult(props: { queryStatus$: BehaviorSubject<QueryStatus> }) {
+  const [isPopoverOpen, setPopover] = useState(false);
+  const [queryStatus, setQueryStatus] = useState<QueryStatus>({ status: 'ready' });
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  useEffect(() => {
+    const subscription = props.queryStatus$.subscribe((status) => {
+      setQueryStatus(status);
+    });
+    return () => subscription.unsubscribe();
+  }, [props.queryStatus$]);
+
+  if (queryStatus.status === 'ready') {
+    return (
+      <EuiButtonEmpty iconSide="left" iconType={'checkInCircleEmpty'} size="xs" onClick={() => {}}>
+        <EuiText size="xs" color="subdued">
+          {'Complete in ' + queryStatus.time + ' ms'}
+        </EuiText>
+      </EuiButtonEmpty>
+    );
+  }
+
+  if (!queryStatus.message || !queryStatus.message.message) {
+    return null;
+  }
+
+  const errorMsg = JSON.parse(queryStatus.message.message);
```

Comment:
should be just `message.message` since we have an `message.error` correct?

Reply:
also careful on parsing . might need to try catch incase the response has bad characters